src/test/ui/nll/issue-98170.rs

```diff
@@ -0,0 +1,13 @@
+pub struct MyStruct<'a> {
+    field: &'a [u32],
+}
+
+impl MyStruct<'_> {
+    pub fn new<'a>(field: &'a [u32]) -> MyStruct<'a> {
+        Self { field }
+        //~^ ERROR lifetime may not live long enough
+        //~| ERROR lifetime may not live long enough
+    }
+}
```

Comment:
Could you also add a test with a trait impl?

Reply:
sure, but it'll be mostly similar to the existing one, see above comment about difficulty of reproducing this error with an elided lifetime in the trait generics